docs/dev/contributing.rst

```diff
@@ -185,22 +277,101 @@ You can install pre-commit by running:
 
```

Comment:
I cant change that line directly but the link does not work (_ missing at the end): and `pre-commit <https://pre-commit.org>`_ to keep everything formatted